nova/virt/vmwareapi/images.py

```diff
@@ -335,7 +335,8 @@ def fetch_image_stream_optimized(context, instance, session, vm_name,
     read_iter = IMAGE_API.download(context, image_ref)
     read_handle = rw_handles.ImageReadHandle(read_iter)
 
-    for i in six.moves.xrange(2):
+    # retry in order to handle conflicts in case of parallel execution (multiple agents)
+    while True:
```

Comment:
Could you limit the number of retries here? Thanks

Reply:
Done.